### DISCUSSION TITLE
How to change port?

### DISCUSSION THREAD

**Discussion by ariefsn** [nuxt/nuxt#16484] [Questions] [answered]
Hi,
how to change port in this v3?
I've try this

```
export default defineNuxtConfig({
  server: {
    port: 1234
  }
})
```

still not work :(

thanks

**Comment by sduduzog** [comment]
edit your script to first pass in the port before starting your dev server i.e. `PORT=1234 nuxt` / `PORT=1234 nuxi dev`

If you're working on multiple machines, try installing `cross-env` and do something like `cross-env PORT=1234 nuxt` so that the script is platform independent

**Reply by ariefsn** [reply]
are we cant using that server block?
I mean, it hardcoded to 1234 but still not work.

**Comment by sduduzog** [comment]
Please see https://github.com/nuxt/framework/issues/893#issuecomment-942211193

**Comment by pi0** [comment, marked as answer]
Hi @ariefsn. You should use `--port` or `PORT` environment variable. Setting from nuxt.config is not supported.

**Reply by ariefsn** [reply]
its working for `--port`,
any idea if it can be set from .env?

thank you...

**Reply by wobsoriano** [reply]
Well you could do something like this in your `package.json` file

```json
{
  "scripts": {
    "dev": "nuxt dev --port=$(grep YOUR_PORT .env | cut -d '=' -f2)"
  }
}
```

**Reply by mikehaertl** [reply]
As a newcomer to Nuxt (V3 in particular) I have to express my surprise that something as simple as setting the host port is not part of the configuration and even requires extra logic if you want to keep it in a `.env` file.

I guess there's good reason for doing that? Or is this yet to be added?

**Reply by pi0** [reply]
It was because how nuxi cli works by starting a listener ahead of time (before loading nuxt and config). But we changed that since I initially answered. I think we can sort out something for config support.

**Reply by mikehaertl** [reply]
Good to hear. Is there maybe an issue that we could follow?

**Reply by fabis94** [reply]
@pi0 hi, I'd also like to track this - is there an issue or maybe this has changed already? (I don't see anything in the docs)

**Reply by vin-ni** [reply]
+1 :)

**Reply by TheAlexLichter** [reply]
Created https://github.com/nuxt/framework/issues/8652 for it
